java-checks/src/main/java/org/sonar/java/checks/tests/AssertJConsecutiveAssertionCheck.java

```diff
@@ -0,0 +1,150 @@
+  public void visitNode(Tree tree) {
+    MethodTree methodTree = (MethodTree) tree;
+    if (hasTestAnnotation(methodTree)) {
+      BlockTree block = methodTree.block();
+      if (block != null) {
+        reportConsecutiveAssertions(block.body());
+      }
+    }
+  }
+
+  private void reportConsecutiveAssertions(List<StatementTree> statements) {
+    ExpressionTree currentArgument = null;
+    MethodInvocationTree currentMit = null;
+    List<MethodInvocationTree> equivalentInvocations = new ArrayList<>();
+
+    for (StatementTree statement : statements) {
+      Optional<MethodInvocationTree> assertThatInvocation = getSimpleAssertSubjectInvocation(statement);
+
+      if (assertThatInvocation.isPresent()) {
+        MethodInvocationTree mit = assertThatInvocation.get();
+        ExpressionTree arg = mit.arguments().get(0);
+        if (currentArgument == null) {
+          currentMit = mit;
+          currentArgument = arg;
+        } else if (areEquivalent(currentArgument, arg)) {
+          equivalentInvocations.add(mit);
+        }
+      } else {
+        // We have something else than an assertion subject between two calls
+        reportIssueIfMultipleCalls(currentMit, equivalentInvocations);
+
+        currentArgument = null;
+        currentMit = null;
+        equivalentInvocations.clear();
+      }
+    }
+
+    reportIssueIfMultipleCalls(currentMit, equivalentInvocations);
+  }
+
+  private static Optional<MethodInvocationTree> getSimpleAssertSubjectInvocation(StatementTree statement) {
+    if (statement.is(Tree.Kind.EXPRESSION_STATEMENT)) {
+      ExpressionTree expression = ((ExpressionStatementTree) statement).expression();
+      if (expression.is(Tree.Kind.METHOD_INVOCATION)) {
+        // First method invocation should be an assertion predicate, if not (incomplete assertion), we will not find anything
+        return getSimpleAssertSubjectInvocation(((MethodInvocationTree) expression).methodSelect());
+      }
+    }
+    return Optional.empty();
+  }
+
+  private static Optional<MethodInvocationTree> getSimpleAssertSubjectInvocation(ExpressionTree expressionTree) {
+    if (expressionTree.is(Tree.Kind.MEMBER_SELECT)) {
+      ExpressionTree memberSelectExpression = ((MemberSelectExpressionTree) expressionTree).expression();
+      if (memberSelectExpression.is(Tree.Kind.METHOD_INVOCATION)) {
+        MethodInvocationTree mit = (MethodInvocationTree) memberSelectExpression;
+        if (ASSERT_THAT_MATCHER.matches(mit)) {
+          return Optional.of(mit);
+        } else if (ASSERTJ_SET_CONTEXT_METHODS.matches(mit)) {
+          return Optional.empty();
+        } else {
+          return getSimpleAssertSubjectInvocation(mit.methodSelect());
+        }
+      }
+    }
+    return Optional.empty();
+  }
+
+  private static boolean areEquivalent(ExpressionTree e1, ExpressionTree e2) {
+    if (e1.is(Tree.Kind.METHOD_INVOCATION)) {
+      // Two method invocation can return different values.
+      return false;
+    }
```

Comment:
Checking `METHOD_INVOCATION` only at the root of an `ExpressionTree` is not enough, we can still have false-positives like:
```
    assertThat(getRandomArray().length).isNotEqualTo(3); // false-positive
    assertThat(getRandomArray().length).isNotEqualTo(4); 
```
I would start from `e1`
for all `MEMBER_SELECT` replace by MemberSelectExpressionTree#expression()
And finally, check `METHOD_INVOCATION`